### PR TITLE
Add warn-starving.lua

### PR DESCRIPTION
**Pull request by PeridexisErrant**
With minor changes to the original version; it doesn't repeat here.
Closes issue #596.

### CODE REVIEW

File: scripts/warn-starving.lua [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    local announcement
+    for i=#units-1, 0, -1 do
+        local unit = units[i]
+        local rraw = findRaceCaste(unit)
+        if rraw and not unit.flags1.dead and not dfhack.units.isOpposedToLife(unit) then
+            local na = checkVariable(unit.counters2.hunger_timer, 75000, 'starving', starvingUnits, unit)
+            na = na or checkVariable(unit.counters2.thirst_timer, 50000, 'dehydrated', dehydratedUnits, unit)
+            na = na or checkVariable(unit.counters2.sleepiness_timer, 150000, 'very drowsy', sleepyUnits, unit)
+            if na then
+                announcement = newAnnouncement
```
Comment (lethosor) [inline scripts/warn-starving.lua]: I'm not sure how this line can work, since newAnnouncement isn't defined anywhere.

Comment (PeridexisErrant) [inline scripts/warn-starving.lua]: Hmm... I tore out some redundant logic, but apparently missed that - [in the gist](https://gist.github.com/Meneth32/ff6fab74d327d4cde84c) it looks like `na` should be an alias for `announcement`.

I'll fix that in a minute.

File: NEWS [hunk 2]
```diff
@@ -12,6 +12,7 @@ DFHack Future
         burial:  sets all unowned coffins to allow burial ("-pets" to allow pets too)
         fix-ster:  changes fertility/sterility of animals or dwarves
         view-item-info: adds information and customisable descriptions to item viewscreens
+        warn-starving:  check for starving, thirsty, or very drowsy units and pasue with warning if any are found
```
Comment (dscorbett) [inline NEWS]: “pasue” → “pause”

File: Readme.rst [hunk 3]
```diff
@@ -2629,6 +2629,12 @@ of items.  Individual descriptions can be added or overridden by a similar
 script ``raw/scripts/more-item-descriptions.lua``.  Both work as sparse lists,
 so missing items simply go undescribed if not defined in the fallback.
 
+warn-starving
+=============
+If any (live) units are starving, very thirsty, or very drowsy, the game will
+be paused and a warnign shown.  Use with the ``repeat`` command for regular
```
Comment (dscorbett) [inline Readme.rst]: “warnign” → “warning”

**Comment by PeridexisErrant** [comment]
Fixed the typos in documentation and glaring variable problems, plus the extra `end` that was breaking things.

There's a more general issue I'm not sure about though:  currently it just prints the message for the first problem it finds, and ignores others.  Is there an elegant way to collect all the messages in a table, then announce them all at once?

**Comment by lethosor** [comment]
Added that and a few other improvements (ability to be used as a module, allowing messages to be re-displayed, etc.)

**Pull request merged**